### PR TITLE
[C#]How to detect a cycle in a Graph.

### LINKED ISSUE #4644: [c#]Cycle Detection in a graph
#### Proposed article introduction
Graphs are used to represent networks. The networks could include paths in a city or telephone network or circuit network. Graphs are also used in social networks like LinkedIn, Facebook. For example, in Facebook, each person is represented with a vertex(or node).
Cyclic graphs have several application in real life scenarios and it is therefore important that we get to understand how to detect it.

#### Key takeaways 
- What is a cycle in a graph.
-  How to detect a cycle using different algorithms giving an advantage of each. 
       1) Depth first search algorithm.
             - Cycle Detection on a directed graph
             - Cycle Detection on a undirected graph
- Understand different applications of cycle detection.

#### Article quality
This article will give a step by step process using well understood and simple visuals on how one can use the different algorithms to detect whether a graph has a cycle. 

#### References
https://www.section.io/engineering-education/graphs-in-data-structure-using-cplusplus/

### PR DESCRIPTION
**Pull request by erixbiz**
#### Formatting and structure checks
- [x] Have you used the correct folder and file structure? - **see [Contribution Guidelines](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/CONTRIBUTING.md)**
- [x] Is your article properly formatted in Markdown? - **see [Github Markdown guide](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)**
- [x] Have you used our preferred heading structure? - **[H3 (###) and up](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet#headers)**
- [x] Does your article follow any of the suggested structure formats? - **see [example formats](https://github.com/section-io/engineering-education/blob/master/CONTRIBUTING.md#example-format-structure)**

- [x] Is your article over 750 words? - **Tool**: [Wordcounter.net](https://wordcounter.net)
- [x] Is the article you are submitting an in-depth and unique article? Does it go beyond what is in the official docs and what is covered in other blog sites. See these articles as [examples](https://github.com/section-engineering-education/engineering-education/blob/master/new_contributors/resources-page.md).
- [x] Does your article provide enough value and detail about your topic? - **Articles should be clear, accurate, and fully explained.**
- [x] Can your article be understood by beginners? - **Assume the audience is smart but has no prior exposure to the common terminology in your article.**

- [x] Have you included a hero image that is 600x400 pixels and have the copyright to use it? - **Tools:** [Pixlr Image Editor](https://pixlr.com/e) for resizing and [Unsplash](https://unsplash.com) for Creative Commons images
      
#### Grammar & plagiarism checks
- [x] Have you spellchecked and grammar checked your article? - **Tools:** [Write&Improve](https://writeandimprove.com/?lang=en-GB) - [ProWritingAid](https://prowritingaid.com/) - [Grammarly](https://grammarly.com) 
- [x] Please place your article through a 3rd party plagiarism checker. We suggest using Quetext, [this tool](https://www.quetext.com/) is free and has a daily limit. **We typically accept articles with 10% or less.**
- [x] Have you checked your article for readability? - **Tool:** [Hemmingway](http://www.hemingwayapp.com)
- [x] Have you added sources for quotes and images that aren't yours?

#### Technical checks
- [x] Are your code snippets properly formatted for syntax highlighting - **see [Syntax guide](https://gohugo.io/content-management/syntax-highlighting)**
- [x] Have you checked your code runs correctly and you've highlighted all necessary dependencies for installation?
- [x] Are the software programs and packages you're highlighting in your article up to date, using current versions and not deprecated?
This fixes #4644

### CODE REVIEW

**Comment by ahmadmardeni1** [comment]
Please always use the template provided by us.
I edited your initial comment and added the template. Check what matches your article.

**Comment by munyite-megzs** [comment]
@hectorkambow  I still don't have a reviewer 5 days later after creating the PR. Are you able to assist me to get someone? Thank you

**Comment by ahmadmardeni1** [comment]
Plagiarism check done ✅

**Comment by munyite-megzs** [comment]
Great
Kind Regards

Margaret Sitati

File: content/articles/graph-cycle-detection-c#/index.md [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+
+[image of a directed-graph](/engineering-education/graph-cycle-detection/directed-graph.png/)
+
+There are undirected graphs as well. An example of this can be Facebook. When you add a friend, there is a connection from you to them. The Vice verse is true. The connections do not have a direction.
+
+The edges can also have weights. We use these weights to represent how strong the connection is, For example, Facebook. When two people communicate, we can put more weight on them. We then use that to show their best friends using the nodes that have the highest weights.
+
+[Image of Weighted Graph](/engineering-education/graph-cycle-detection/weightedgraph.png/)
+
+With this brief description, we can get to know what a cycle is and how to detect that in a graph.
```
Comment (jumamiller) [inline content/articles/graph-cycle-detection-c#/index.md:21]: Could we have this part moved into the article's body while only briefly introducing this section? I mean it's too long to be an introduction!

File: content/articles/graph-cycle-detection-c#/index.md [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+
+- Understand What is a Cycle in a Graph.
+
+- Understand How to detect a Cycle. 
+
+      1) Depth-first search algorithm.
+
+            - Cycle Detection on a directed graph
+
+            - Cycle Detection on an undirected graph
```
Comment (jumamiller) [inline content/articles/graph-cycle-detection-c#/index.md:45]: Combining both uppercase and lowercase on a sentence? Combining both uppercase and lowercase on a sentence?

File: content/articles/graph-cycle-detection-c#/index.md [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+
+1. Open Visual Studio. Create a Console Application called GraphCycleDetect.
+
+2. In the main method we create a jagged array. This is what will show how vertexes are connected.
+
+3. We Declare a variable to hold the number of vertexes.
+
+4. We Call the MakeGraph function. Pass the number of vertexes and the graph as the parameters to that function. Below is the code that implements steps 1 to 4.
+
+```c#
```
Comment (jumamiller) [inline content/articles/graph-cycle-detection-c#/index.md:113]: Ensure all methods are included take the following format: `MakeGraph()`

File: content/articles/graph-cycle-detection-c#/index.md [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+
+               new int[]{ 1,5},
+
+               new int[]{ 2,6},
+
+               new int[]{ 2,7},
+
+               new int[]{ 3,4},
+
+               new int[]{ 4,1},
```
Comment (jumamiller) [inline content/articles/graph-cycle-detection-c#/index.md:137]: Explain all code snippets!

File: content/articles/graph-cycle-detection-c#/index.md [hunk 5]
```diff
@@ -0,0 +1,431 @@
...
+
+1. It is used a lot in distributed message-based algorithms.
+
+2. Used in concurrent systems to detect deadlocks.
+
+3. Used in cryptographic systems. It determines the keys of a message that can map that same message to the same encrypted value.
+
+### Conclusion
+
+I hope that after going through this article, the concept of cycle detection is clearer to you. Happy Coding!
```
Comment (jumamiller) [inline content/articles/graph-cycle-detection-c#/index.md:431]: Conclusion is meant to cover what you discussed on the article!

**Review by jumamiller — CHANGES_REQUESTED** [review]
Hello, work on the above suggestions!

**Comment by erixbiz** [comment]
@jumamiller  done with the changes requested

File: content/articles/graph-cycle-detection-c#/index.md [hunk 6]
```diff
@@ -0,0 +1,462 @@
+A graph is a non-linear data structure that consists of:-
+
+1. Some grouped vertices are also called nodes.
+
+2. Some ordered pairs of the form (u, v) are called edges.
+
+
```
Comment (jumamiller) [inline content/articles/graph-cycle-detection-c#/index.md:7]: Where is the intro?

File: content/articles/graph-cycle-detection-c#/index.md (hunk 3 above)
Comment (jumamiller) [reply content/articles/graph-cycle-detection-c#/index.md:120]: Not fixed!

File: content/articles/graph-cycle-detection-c#/index.md [hunk 7]
```diff
@@ -0,0 +1,462 @@
...
+            // those nodes from this path. This is done recursively.
+           path[start] = false;
+            // If we did not find a cycle, we return false.
+           return false;
+
+       }
+
+```
+
+### OUTPUT
```
Comment (jumamiller) [inline content/articles/graph-cycle-detection-c#/index.md:291]: Output shouldnt be heading!

**Review by jumamiller — CHANGES_REQUESTED** [review]
@erixbiz you have not polished the article. It lacks the general flow, no introduction, the suggested changes were not edited.

I'll close this article for now, kindly go thro' past articles and get the general flow!Ping me here when done.

**Comment by erixbiz** [comment]
@jumamiller sorry for that. will go through it thoroughly and let you know once done

**Pull request closed without merging**